File: app/src/main/java/com/bq/zowi/models/commands/BaseCommand.java
package com.bq.zowi.models.commands;

import com.bq.zowi.models.commands.Command;
import java.util.Arrays;

/* JADX INFO: loaded from: classes.dex */
public abstract class BaseCommand implements Command {
    private Command.Action action;
    private Command.Direction direction;
    protected Long duration;

    @Override // com.bq.zowi.models.commands.Command
    public Command.Action getAction() {
        return this.action;
    }

    @Override // com.bq.zowi.models.commands.Command
    public void setAction(Command.Action action) {
        if (!Arrays.asList(getAllowedActions()).contains(action)) {
            throw new IllegalArgumentException("The action " + action.toString() + " is not valid here.");
        }
        this.action = action;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Direction getDirection() {
        return this.direction;
    }

    @Override // com.bq.zowi.models.commands.Command
    public void setDirection(Command.Direction direction) {
        if (!Arrays.asList(getAllowedDirections()).contains(direction)) {
            throw new IllegalArgumentException("The direction " + direction.toString() + " is not valid for action " + getAction().toString());
        }
        this.direction = direction;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Long getDuration() {
        return this.duration;
    }

    @Override // com.bq.zowi.models.commands.Command
    public void setDuration(long duration) {
        if (!Arrays.asList(getAllowedDurations()).contains(Long.valueOf(duration))) {
            throw new IllegalArgumentException("The duration " + duration + " is not valid for action " + getAction().toString());
        }
        this.duration = Long.valueOf(duration);
    }

    @Override // com.bq.zowi.models.commands.Command
    public String getId() {
        return getAction().getId();
    }
}

File: zowi-core/src/main/kotlin/com/bq/zowi/models/commands/BaseCommand.kt
package com.bq.zowi.models.commands

abstract class BaseCommand : Command {
    private var action: Command.Action? = null
    private var direction: Command.Direction? = null
    protected var _duration: Long? = null

    override fun getAction(): Command.Action = action!!

    override fun setAction(action: Command.Action) {
        require(action in getAllowedActions()) { "The action $action is not valid here." }
        this.action = action
    }

    override fun getDirection(): Command.Direction? = direction

    override fun setDirection(direction: Command.Direction) {
        require(direction in getAllowedDirections()) { "The direction $direction is not valid for action ${getAction()}" }
        this.direction = direction
    }

    override fun getDuration(): Long? = _duration

    override fun setDuration(duration: Long) {
        require(duration in getAllowedDurations()) { "The duration $duration is not valid for action ${getAction()}" }
        this._duration = duration
    }

    protected fun setDurationRaw(duration: Long) {
        this._duration = duration
    }

    override fun getId(): String = getAction().id
}
